src/k8s/pkg/component/loadbalancer.go

```diff
@@ -39,6 +39,25 @@ func UpdateLoadBalancerComponent(ctx context.Context, s snap.Snap, isRefresh boo
 		return fmt.Errorf("failed to enable load-balancer component: %w", err)
 	}
 
+	// Wait for cilium CRDs to be available.
+	k8sClient, err := k8s.NewClient(s)
+	if err != nil {
+		return fmt.Errorf("failed to create k8s client: %w", err)
+	}
+	control.WaitUntilReady(ctx, func() (bool, error) {
+		serverGroups, err := k8sClient.ListServerGroups()
+		if err != nil {
+			return false, fmt.Errorf("failed to check CRDs: %w", err)
+		}
+
+		for _, sg := range serverGroups.Groups {
+			if sg.Name == "cilium.io" {
+				return true, nil
+			}
+		}
+		return false, nil
```

Comment:
let's specifically check for the crds we care about. use `clientset.Discovery().ServerResourcesForGroupVersion()` and check for the l2 announcement or bgp peering policy crds

Reply:
thanks - forgot to push the changes yesterday.